packages/web3-utils/src/socket_provider.ts

```diff
@@ -331,6 +349,27 @@ export abstract class SocketProvider<
 		this._onDisconnect(disconnectCode, data);
 	}
 
+	/**
+	 * Safely disconnects the socket, async and waits for request size to be 0 before disconnecting
+	 * @param code - The code to be sent to the server
+	 * @param data - The data to be sent to the server
+	 */
+	public async safeDisconnect(code?: number, data?: string) {
+		const checkQueue = async () => 
+			new Promise(resolve => {
+				const interval = setInterval(() => {
+					if (this.getPendingRequestQueueSize() === 0 && this.getSentRequestsQueueSize() === 0) {
+						clearInterval(interval);
+						resolve(true);
+					}
+				}, 1000)
```

Comment:
I think,`1000` should be an options param in the safeDisconnect method. by default 1000

Reply:
updated, thanks